app/src/main/java/com/mandysoftware/livrenouveau/DefinitionActivity.kt

```diff
@@ -0,0 +1,183 @@
+package com.mandysoftware.livrenouveau
+
+import android.app.Activity
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.view.MenuItem
+import android.widget.SearchView
+import android.widget.TextView
+import android.widget.Toast
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.mandysoftware.wordutil.CompositeDictionary
+import com.mandysoftware.wordutil.CompositeDictionary.DictionaryCallback
+//import org.xwiki.rendering.wikimodel.xwiki.xwiki20.XWikiParser
+import java.io.File
+import java.lang.Exception
+
+class DefinitionActivity : AppCompatActivity(), DictionaryCallback {
+    val TAG = "DefinitionActivity"
+    var word = ""
+    var tvDefinition : TextView? = null
+
+    var dictionary : CompositeDictionary? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_definition)
+
+
+
+        val storageDir = filesDir
+        val dictionaryDir = File(storageDir, "dictionary")
+        tvDefinition = findViewById(R.id.tvDefinition)
+
+
+
+
+        word = intent.getStringExtra("word")!!
+
+
+        // INIT is the same as LOAD_OFFLINE, but INIT has an extra
+        // preparation step of checking to make sure the dictionary
+        // list exists, and, if it doesn't exist, downloading it
+        // from the server.
+        dictionary = CompositeDictionary(dictionaryDir)
+        dictionary!!.addDictionaryCallback(this)
+        dictionary!!.postCommand(CompositeDictionary.Command(CompositeDictionary.CommandName.INIT))
+        dictionary!!.postCommand(
+            CompositeDictionary
+                .Command(CompositeDictionary.CommandName.LOOKUP, word))
+
+
+
+
+        // Set up SearchView listener
+        val di = dictionary
+
+        // Get a reference to the action bar
+        supportActionBar?.apply {
+            // Enable the "up" button on the action bar
+            setDisplayHomeAsUpEnabled(true)
+
+        }
+
+
+
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        // Handle "up" button clicks
+        if (item.itemId == android.R.id.home) {
+            val intent = Intent(this@DefinitionActivity, SearchActivity::class.java)
+            intent.putExtra("word", word)
+            setResult(Activity.RESULT_OK, intent)
+            finish()
```

Comment:
Take these 4 lines of code and refactor them into their own function.

Call this function `goBack()`, and let it be a `void goBack()` type.

Reason for this is later on, we'll want to expand on what the "go back" feature does. Having it in its own function will allow us to do so cleanly.